crypto/evp/evp_lib.c

```diff
@@ -217,13 +217,12 @@ int EVP_CIPHER_type(const EVP_CIPHER *ctx)
 
 int EVP_CIPHER_block_size(const EVP_CIPHER *cipher)
 {
-    if (cipher->prov != NULL) {
-        if (cipher->blocksize != NULL)
-            return cipher->blocksize();
-        /* We default to a block size of 1 */
-        return 1;
-    }
-    return cipher->block_size;
+    int v = cipher->block_size;
```

Comment:
same comments as for ```EVP_CIPHER_CTX_set_padding```

Reply:
No.  I would rather not have to repeat `cipher-block_size` three times.  Also, saving it in a temporary variable is a safety measure against possible bugs in `evp_do_param` or other functions down the line, so they don't get direct access to `cipher->block_size` and get to trample all over that structure.